Assets/Scripts/Contract/ContractController.cs

```diff
@@ -69,28 +81,25 @@ public void LogAllContracts()
         }
     }
 
-    //progress contract if workers are assigned
-    public void ContractProgression(string guid)
-    {
-        Contract c = contracts[guid];
-
-        if (c.GetAssignedWorkers() != null)
-        {
-            float effort = 1 * (c.GetAmountOfAssignedWorkers());
-            c.SetTotalEffort(effort);
-            Debug.Log("progressing" + c.GetTotalEffort());
-        }
-    }
+    public DateTime timeOfLastTick = DateTime.Now;
+    public int TICK_THRESHOLD = 100000;
 
-    //check for contract completion
-    public void ContractCompletion(string guid)
+    public void Tick()
     {
-        Contract c = contracts[guid];
+        if (DateTime.Now.Ticks - timeOfLastTick.Ticks >= TICK_THRESHOLD) {
+            Debug.Log("Ticking");
+            foreach (KeyValuePair<string, Contract> entry in contracts) 
+            {
+                if (entry.Value.IsActive()) {
+                    entry.Value.IncrementWork(entry.Value.GetAmountOfAssignedWorkers());
+                    // subtract money for every worker we find
```

Comment:
This comment looks like it's describing paying workers, but the code looks like it's decrementing the amount of work remaining on the contract.

I think that worker pay should be handled by the EmployeeController since there are too many edge cases if the ContractController handles pay. (e.g. not assigned to a contract, assigned to more than one contract)

Reply:
Yeah, I left a comment reminding me to come back to it. I wanted to come back to this once we had the resource system created.